kod/object/item.kod

```diff
@@ -613,6 +613,52 @@ messages:
       return 1;
    }
 
+   GetItemRarityGrade(what = $)
+   "Returns the appropriate text coloring tier for the item."
+   {
+      local irarity, i;
+      irarity = ITEM_RARITY_GRADE_NORMAL;
+
+      % Provide default value if not an item
+      if NOT IsClass(self,&Item)
+      {
+         return ITEM_RARITY_GRADE_NORMAL;
+      }
+
+      if Send(self,@IsCursed)
+      {
+         return ITEM_RARITY_GRADE_CURSED;
+      }
+
+      % Static class rarities
+      if IsClass(self,&helm)               % Magic Spirit Helm
+      {
+         irarity = ITEM_RARITY_GRADE_LEGENDARY;
+      }
+
+      % Want to refactor to check a list of classes to see if 
+      % the item class is in there and assign appropriately
+      if IsClass(self,&KraananCharm)       % Steel Torc
+         OR IsClass(self,&FarenCharm)      % Brass Necklace
+         OR IsClass(self,&QorCharm)        % Monacle
+         OR IsClass(self,&RiijaCharm)      % Feathered Choker
+         OR IsClass(self,&ShalilleCharm)   % Diamond Pendant
+         OR IsClass(self,&MagicBow)
+         OR IsClass(self,&NeruditeBow)
+         OR IsClass(self,&GoldSword)
+      {
+         irarity = ITEM_RARITY_GRADE_RARE;
+      }
+
+      if IsClass(self,&RiijaSword)         % Feathered Choker
```

Comment:
Same thing here.

It sounds like perhaps you want a function GetRarity that takes no arguments, and returns a class variable, itself defaulted to ITEM_RARITY_GRADE_NORMAL in Item.  This function could check if the item is cursed.  Subclasses would override the class variable if their rarity isn't NORMAL.

Reply:
Thank you, I refactored the rarity logic based on this feedback.  It's starting to make sense to me.  This works really well for item classes.

I would like to, if possible, implement 2 other ways items could have a higher rarity grade than "normal."
- some logic to count up the item attributes and return a rarity of magic (1 attribute), rare (2+ attributes)
- a way for administrators or DMs to assign the rarity grade to something they make

I will try and modify `GetRarity` to do allow for those 2 conditions.